elements/drag-drop-elements/pl-drag-drop-element.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-env jquery, browser */
+
+const TABWIDTH = 50;      // defines how many px the answer block is indented by, when the student
+                          // drags and indents a block
+const INDENT_OFFSET = 5;  // For aesthetic, all answer blocks are offseted to the right by
+                          // 5px, so the answer tiles are not directly touching the dropzone margins
+const MAX_INDENT = 4;     // defines the maximum number of times an answer block can be indented
+
+function set_answer(event) {
+    // We only care about when this function is fired
+    // from an ANSWER DROPZONE, aka dropzones with yellow backgrounds 
+    var textfield_name = event.target.getAttribute('name');
+    var dom_objs = $('#' + textfield_name + '-dropzone').children();
+    var student_answers = [];
+    var indents = [];
+    var answer_json = {'answers': [], 'answer_indent': []};
+    for (var i = 0; i < dom_objs.length; i++) {
+        if (!$(dom_objs[i]).hasClass('info')){
+            var answer_text = dom_objs[i].getAttribute('string');
+            var answer_indent = parseInt($(dom_objs[i]).css('marginLeft').replace('px', ''));
+            answer_indent = Math.round((answer_indent - INDENT_OFFSET) / TABWIDTH); // get how many times the answer is indented
+            student_answers.push(answer_text);
+            indents.push(answer_indent.toString());
+        }
+    }
+
+    textfield_name = event.target.getAttribute('name');
+    if (textfield_name === null){
+        return;
+    }
+    textfield_name = '#' + textfield_name + '-input';
+    answer_json.answers = student_answers;
+    answer_json.answer_indent = indents;
+    $(textfield_name).val(JSON.stringify(answer_json));
+}
+
+
+function update_indent(leftDiff, id, ui) {
```

Comment:
Blocks can't be unindented after they've been dragged back to the source area. We should either reset the indentation once they're dragged back, or at least allow unindentation in the source area.

![image](https://user-images.githubusercontent.com/2503508/92771201-4ea3f780-f360-11ea-8728-d9f8b2b6e312.png)

Reply:
yes, I think I tried to make the blocks unindented once dragged back to the original table by setting the style back to 5px, but ill look into the bugs

Reply:
fixed in latest commit